server/core/region.go

```diff
@@ -630,6 +630,22 @@ func (r *RegionsInfo) GetRegions() []*RegionInfo {
 	return regions
 }
 
+// GetStoreRegions gets all RegionInfo with a given storeID
+func (r *RegionsInfo) GetStoreRegions(storeID uint64) []*RegionInfo {
+	var regions []*RegionInfo
+	if leaders, ok := r.leaders[storeID]; ok {
+		for _, region := range leaders.m {
+			regions = append(regions, region.Clone())
```

Comment:
Seems no need to `Clone` any more after the refactor of `RegionInfo`. /cc @nolouch